.github/workflows/backend-lint.yml

```diff
@@ -0,0 +1,45 @@
+name: backend-lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  go-mod-tidy:
+    strategy:
+      matrix:
+        go-version: [1.16.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Set up Golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    ## Run go mod tidy and then check if there was a diff. If so, fail the check.
+    - name: Check go mod tidy diff
+      run: if [ $(go mod tidy && git diff | wc -l) -gt 0 ]; then git diff && exit 1; fi
```

Comment:
for those others curious about this line, there's no other good way of doing this: https://github.com/golang/go/issues/27005